quasiquotes3/shared/src/main/scala/scala/meta/quasiquotes/Api.scala

```diff
@@ -0,0 +1,196 @@
+    transparent inline def template(inline args: Any*): Template = ${ ReificationMacros.templateImpl('sc, 'args) }
+    transparent inline def mod(inline args: Any*): Mod = ${ ReificationMacros.modImpl('sc, 'args) }
+    transparent inline def enumerator(inline args: Any*): Enumerator = ${ ReificationMacros.enumeratorImpl('sc, 'args) }
+    transparent inline def importer(inline args: Any*): Importer = ${ ReificationMacros.importerImpl('sc, 'args) }
+    transparent inline def importee(inline args: Any*): Importee = ${ ReificationMacros.importeeImpl('sc, 'args) }
+    transparent inline def source(inline args: Any*): Source = ${ ReificationMacros.sourceImpl('sc, 'args) }
+  }
+
+  // unapply methods
+  extension (stringContext: scala.StringContext)
+    @annotation.compileTimeOnly(".q should not be called directly. Use q\"...\" string interpolation.")
+    def q: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".param should not be called directly. Use param\"...\" string interpolation.")
+    def param: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".t should not be called directly. Use t\"...\" string interpolation.")
+    def t: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".tparam should not be called directly. Use tparam\"...\" string interpolation.")
+    def tparam: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".p should not be called directly. Use p\"...\" string interpolation.")
+    def p: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".init should not be called directly. Use init\"...\" string interpolation.")
+    def init: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".self should not be called directly. Use self\"...\" string interpolation.")
+    def self: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".template should not be called directly. Use template\"...\" string interpolation.")
+    def template: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".mod should not be called directly. Use mod\"...\" string interpolation.")
+    def mod: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".enumerator should not be called directly. Use enumerator\"...\" string interpolation.")
+    def enumerator: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".importer should not be called directly. Use importer\"...\" string interpolation.")
+    def importer: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".importee should not be called directly. Use importee\"...\" string interpolation.")
+    def importee: QuasiquoteUnapply = ???
+    @annotation.compileTimeOnly(".source should not be called directly. Use source\"...\" string interpolation.")
+    def source: QuasiquoteUnapply = ???
+  
+  extension (inline sc: QuasiquoteUnapply)
+    transparent inline def unapplySeq(scrutinee: Any): Option[Seq[Any]] =
+      ${ ReificationMacros.unapplyImpl('sc, 'scrutinee) }
+
+
+  // parsers
+  object XTensionQuasiquoteTerm {
```

Comment:
How are those parsers used? I don't see any explicit usage.

Reply:
In `ReificationMacros` in `instantiateParser()`, they are classloaded and invoked (just like in scala 2). In scala 2 this boilerplate was hidden inside classes with a macro annotation